core/lib/default_da_clients/src/no_da/client.rs

```diff
@@ -0,0 +1,36 @@
+use std::fmt::Debug;
+
+use async_trait::async_trait;
+use zksync_da_client::{
+    types::{DAError, DispatchResponse, InclusionData},
+    DataAvailabilityClient,
+};
+
+/// A no-op implementation of the `DataAvailabilityClient` trait, that doesn't store the pubdata.
+#[derive(Clone, Debug, Default)]
+pub struct NoDAClient;
+
+impl NoDAClient {
+    pub fn new() -> Self {
+        NoDAClient {}
+    }
+}
+
+#[async_trait]
+impl DataAvailabilityClient for NoDAClient {
+    async fn dispatch_blob(&self, _: u32, _: Vec<u8>) -> Result<DispatchResponse, DAError> {
+        Ok(DispatchResponse::default())
+    }
```

Comment:
it is called for noDA Validium because we don't want `da_dispatcher` to know anything about the client implementation, so just doing the same-flow-for-everyone keeps the code simpler in this case